test(transformer): add nestjs e2e test

Dunqing opened the pull request:
Add a NestJS e2e test to avoid breaking legacy decorator + `emitDecoratorMetadata` transformations. Code copied from the NestJS template.

Boshen commented:
### Merge activity

* **Apr 27, 7:58 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
test(transformer): add nestjs e2e test (#10634)

Add a NestJS e2e test to avoid breaking legacy decorator + `emitDecoratorMetadata` transformations. Code copied from the NestJS template.
